mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -4265,3 +4265,90 @@ fn test_utxo_validate_valid_and_invalid_pubkey() {
     assert!(coin.validate_other_pubkey(&[1u8; 20]).is_err());
     assert!(coin.validate_other_pubkey(&[1u8; 8]).is_err());
 }
+
+#[test]
+fn test_block_header_utxo_loop() {
+    use crate::utxo::utxo_builder::{block_header_utxo_loop, BlockHeaderUtxoLoopExtraArgs};
+    use futures::future::{Either, FutureExt};
+
+    static mut CURRENT_BLOCK_COUNT: u64 = 150;
+
+    ElectrumClient::get_block_count
+        .mock_safe(move |_| MockResult::Return(Box::new(futures01::future::ok(unsafe { CURRENT_BLOCK_COUNT }))));
+
+    BlockHeaderUtxoLoopExtraArgs::default.mock_safe(move || {
+        MockResult::Return(BlockHeaderUtxoLoopExtraArgs {
+            chunk_size: 100,
+            error_sleep: 5.,
+            success_sleep: 1.,
+        })
+    });
+
+    let block_header_storage = BlockHeaderStorageForTests::new(TEST_COIN_NAME.to_string());
+    BlockHeaderStorage::new_from_ctx.mock_safe(move |_, _| {
+        MockResult::Return(Ok(BlockHeaderStorage {
+            inner: Box::new(block_header_storage.clone()),
+        }))
+    });
+
+    let ctx = MmCtxBuilder::new().into_mm_arc();
+    let priv_key_policy = PrivKeyBuildPolicy::IguanaPrivKey(&[1u8; 32]);
+
+    let mut args = ElectrumBuilderArgs::default();
+    args.collect_metrics = false;
+    args.spawn_ping = false;
+
+    let servers: Vec<_> = [
+        "electrumx1.cointest.com:50001",
+        "electrumx2.cointest.com:50001",
+        "electrumx3.cointest.com:50001",
+    ]
+    .iter()
+    .map(|server| json!({ "url": server,"disable_cert_verification":true }))
+    .collect();
+    let servers = servers.into_iter().map(|s| json::from_value(s).unwrap()).collect();
+    let conf = json!({"coin":"WHIVE","asset":"WHIVE","rpcport":8923,"enable_spv_proof": false});
+    let req = json!({ "method": "electrum", "servers": servers });
+
+    let params = UtxoActivationParams::from_legacy_req(&req).unwrap();
+    let builder = UtxoArcBuilder::new(&ctx, "WHIVE", &conf, &params, priv_key_policy, UtxoStandardCoin::from);
+    let client = block_on(builder.electrum_client(AbortableQueue::default(), args, servers)).unwrap();
```

Comment:
I believe that there is no reason to create the `electrum_client` manually since `build_utxo_fields` will create it again.
So you can do
```rust
let builder = UtxoArcBuilder::new(&ctx, "WHIVE", &conf, &params, priv_key_policy, UtxoStandardCoin::from);
let arc: UtxoArc = block_on(builder.build_utxo_fields()).unwrap().into();
let client = match arc.rpc_client {
  ::Electrum(electrum) => electrum,
  ::Native(_) => unreachable!(),
};
```